eng/references.targets

```diff
@@ -14,34 +15,35 @@
     </ProjectReference>
   </ItemDefinitionGroup>
 
+  <!-- Set the corresponding CoreLib runtime configuration. -->
   <ItemGroup Condition="'@(ProjectReference)' != ''">
     <_coreLibProjectReference Include="@(ProjectReference->WithMetadataValue('Identity', '$(CoreLibProject)'))" />
-    <ProjectReference Update="@(_coreLibProjectReference)"
-                      Private="false">
+    <ProjectReference Update="@(_coreLibProjectReference)">
       <SetConfiguration Condition="'$(RuntimeFlavor)' == 'CoreCLR' and
                                    '$(Configuration)' != '$(CoreCLRConfiguration)'">Configuration=$(CoreCLRConfiguration)</SetConfiguration>
       <SetConfiguration Condition="'$(RuntimeFlavor)' == 'Mono' and
                                    '$(Configuration)' != '$(MonoConfiguration)'">Configuration=$(MonoConfiguration)</SetConfiguration>
     </ProjectReference>
-    <!-- If a CoreLib ProjectReference is present, make all P2P assets non transitive. -->
-    <ProjectReference Update="@(ProjectReference->WithMetadataValue('PrivateAssets', ''))"
-                      PrivateAssets="all"
-                      Condition="'$(IsSourceProject)' == 'true' and '@(_coreLibProjectReference)' != ''" />
   </ItemGroup>
 
-  <!-- Make shared framework assemblies not app-local (non private). -->
-  <Target Name="UpdateProjectReferencesWithPrivateAttribute"
-          AfterTargets="AssignProjectConfiguration"
-          BeforeTargets="PrepareProjectReferences"
-          Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' and
-                     ('$(IsTestProject)' == 'true' or '$(IsTestSupportProject)' == 'true') and
-                     '@(ProjectReferenceWithConfiguration)' != ''">
-    <ItemGroup>
-      <ProjectReferenceWithConfiguration PrivateAssets="all"
-                                         Private="false"
-                                         Condition="$(NetCoreAppLibrary.Contains('%(Filename);')) and '%(ProjectReferenceWithConfiguration.Private)' == ''" />
-    </ItemGroup>
-  </Target>
+  <!-- Mark shared framework assemblies as non-transitive (privateassets=all) and non app-local (private=false). -->
+  <ItemGroup Condition="'@(ProjectReference)' != '' and
+                        '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and
+                        '$(TargetFrameworkVersion)' == 'v$(NetCoreAppCurrentVersion)'">
+    <_ProjectReferenceWithOriginalIdentity Include="@(ProjectReference)"
+                                           OriginalIdentity="%(Identity)" />
+    <_projectReferenceWithFilename Include="@(_ProjectReferenceWithOriginalIdentity->Metadata('Filename'))" />
+
+    <_projectReferenceExcludedWithFilename Include="@(_projectReferenceWithFilename)"
+                                           Exclude="@(NetCoreAppLibrary)" />
+    <_projectReferenceIncludedWithFilename Include="@(_projectReferenceWithFilename)"
```

Comment:
nit: naming here, consider something like `_frameworkProjectReferences` as it might be more descriptive than `_projectReferenceIncludedWithFilename`